# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xf86-video-sis/hotfix-x32.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Custom memcpy assembly not x32 compatible, ...

--- xf86-video-sis-0.12.0/src/sis_memcpy.c.vanilla	2021-11-07 16:42:37.256092451 +0100
+++ xf86-video-sis-0.12.0/src/sis_memcpy.c	2021-11-07 16:42:51.500093186 +0100
@@ -1149,7 +1149,7 @@
 {
 #if defined(__i386__) && defined(SiS_canBenchmark)
     return(SiSVidCopyInitGen(pScreen, MCFunctions_i386, UMemCpy, from));
-#elif (defined(__AMD64__) || defined(__amd64__) || defined(__x86_64__)) && defined(SiS_canBenchmark)
+#elif (defined(__AMD64__) || defined(__amd64__) || defined(__x86_64__)) && defined(SiS_canBenchmark) && !defined(_ILP32)
     return(SiSVidCopyInitGen(pScreen, MCFunctions_AMD64, UMemCpy, from));
 #else /* Other cases: Use libc memcpy() */
     *UMemCpy = SiS_libc_memcpy;
